osu.Game.Tournament/Screens/SetupScreen.cs

```diff
@@ -186,15 +189,70 @@ public bool Failing
                         Anchor = Anchor.CentreLeft,
                         Origin = Anchor.CentreLeft,
                     },
-                    button = new TriangleButton
+                    FlowContainer = new FillFlowContainer
                     {
                         Anchor = Anchor.CentreRight,
                         Origin = Anchor.CentreRight,
-                        Size = new Vector2(100, 30),
-                        Action = () => Action?.Invoke()
-                    },
+                        AutoSizeAxes = Axes.Both,
+                        Spacing = new Vector2(10, 0),
+                        Children = new Drawable[]
+                        {
+                            Button = new TriangleButton
+                            {
+                                Size = new Vector2(100, 30),
+                                Action = () => Action?.Invoke()
+                            }
+                        }
+                    }
                 }
             };
         }
+
+        private class ResolutionSelector : ActionableInfo
+        {
+            private const int height_min_allowed_value = 480;
+            private const int height_max_allowed_value = 2160; // 4k
+            public new Action<int> Action;
+
+            private OsuNumberBox numberBox;
+
+            protected override Drawable CreateComponent()
+            {
+                var drawable = base.CreateComponent();
+                FlowContainer.Insert(0, numberBox = new OsuNumberBox
+                {
+                    Width = 100
+                });
+                FlowContainer.SetLayoutPosition(Button, 1);
+
+                base.Action = () =>
+                {
+                    if (numberBox.Text.Length > 0)
+                    {
+                        // box contains text
+                        if (!int.TryParse(numberBox.Text, out var number))
+                        {
+                            // at this point, the only reason we can arrive here is if the input number was too big to parse into an int
+                            // so clamp to max allowed value
+                            number = height_max_allowed_value;
+                        }
+                        else
+                        {
+                            number = Math.Clamp(number, height_min_allowed_value, height_max_allowed_value);
+                        }
+
+                        // in case number got clamped, reset number in numberBox
+                        numberBox.Text = number.ToString();
+
+                        Action?.Invoke(number);
+                    }
+                    else
+                    {
+                        // TODO: input box was empty, give user feedback? do nothing?
+                    }
```

Comment:
You could disable the button if you really wanted in this branch (remembering to re-enable it once input is non-empty again). But that's about as far as I'd go.

Reply:
well this code runs when you click the button, so that wouldnt make a ton of sense.

guess I'll just keep it at just doing nothing at all. I guess should be obvious enough that if you want that button to do something you should enter a value. Altho while im typing this I get an idea: give the inputbox focus as to hint the user they should type something there. Is there a way to set focus to an inputbox in code? I tried to look for something but couldnt find anything 🤔

Reply:
It would visually indicate that the button cannot be clicked due to lack of input in the text box. But again, this doesn't really matter this much at all. As far as I'm concerned it can just do nothing, I'd just remove the todo.